vm/src/dictdatatype.rs

```diff
@@ -573,13 +606,35 @@ impl<T: Clone> Dict<T> {
         Ok(removed)
     }
 
-    pub fn pop_back(&self) -> Option<(PyObjectRef, T)> {
-        let mut inner = self.write();
-        inner.entries.pop().map(|entry| {
-            inner.used -= 1;
-            inner.indices[entry.index] = IndexEntry::DUMMY;
-            (entry.key, entry.value)
-        })
+    pub fn pop_back(&self, vm: &VirtualMachine) -> Option<(PyObjectRef, T)> {
+        let inner = self.read();
+        let mut idx: i64 = inner.nentries as i64 - 1;
+
+        while idx >= 0 && inner.entries.get(idx as usize).unwrap().is_none() {
+            idx -= 1;
+        }
```

Comment:
is this nessessary if `nentries` is being keeped maintained well?

Reply:
since pop(key) can make entry to None in the middle of the entries. it is necessary

Reply:
So, what happens if we repeat to delete last item and add new item one by one? Does it infinitely grow instead of reusing the last entry?

Reply:
if we use pop(key) or delete to delete the last item, it is infinitely grow. but when inserting, the resize calls if the should_resize returns true. this will remove None entries and the infinite grow would not happen that much i think. I can make the delete function to detect whether the deleted item was the last entry, but this can also happen when deleting the first and insert again. I think it's a problem about resize function.

Reply:
@youknowone do i have to fix resize also? or left to it another issue?

Reply:
I am not sure about the best solution. At the same time, that also look like not a strict purpose of this PR. So vote for leaving it for later.